LoRaEngine/modules/LoRaWanNetworkSrvModule/LoraTools/LoRaMessage/LoRaPayloadJoinAccept.cs

```diff
@@ -98,6 +98,9 @@ public LoRaPayloadJoinAccept(string netId, byte[] devAddr, byte[] appNonce, byte
             }
         }
 
+        public LoRaPayloadJoinAccept(ReadOnlyMemory<byte> inputMessage, AppKey appKey) : this(inputMessage.ToArray(), appKey)
+        { }
+
         public LoRaPayloadJoinAccept(byte[] inputMessage, AppKey appKey)
```

Comment:
```suggestion
        private LoRaPayloadJoinAccept(byte[] inputMessage, AppKey appKey)
```

This constructor should be made `private` to avoid two different ways of constructing a class instance where one constructor makes a copy of the data and the other doesn't. Having two different policies could lead to subtle bugs since a `ReadOnlyMemory<>` doesn't by itself guarantee that the underlying array cannot change. It's up to the class implementation to take a copy of the input and expose that in an immutable manner.